fitbenchmarking/controllers/kmpfit_controller.py

```diff
@@ -0,0 +1,88 @@
+"""
+Implements a controller for the Kmpfit software.
+"""
+
+from kapteyn import kmpfit
+from fitbenchmarking.controllers.base_controller import Controller
+
+
+class KmpfitController(Controller):
+    """
+    Controller for Kmpfit
+    """
+
+    algorithm_check = {
+        'all': ['lm-lsqr'],
+        'ls': ['lm-lsqr'],
+        'deriv_free': [],
+        'general': [],
+        'simplex': [],
+        'trust_region': [],
+        'levenberg-marquardt': ['lm-lsqr'],
+        'gauss_newton': [],
+        'bfgs': [],
+        'conjugate_gradient': [],
+        'steepest_descent': [],
+        'global_optimization': []}
+
+    jacobian_enabled_solvers = ['lm-lsqr']
+
+    def __init__(self, cost_func):
+        """
+        Initialises variables used for temporary storage.
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+        """
+        super().__init__(cost_func)
+        self.result = None
+        self._status = None
+        self._popt = None
+        self.kmpfit_fitter = kmpfit.Fitter
+        self.kmpfit_object = None
+
+    def kmpfit_residuals(self, p, data):
+        """
+        Residuals for Kmpfit
+        """
+        _, _ = data
+        return self.cost_func.eval_r(p)
+
+    def kmpfit_jacobians(self, p, data, dflags):
+        """
+        Jacobians for Kmpfit
+        """
+        _, _ = data
+        _ = dflags
+        jac = self.cost_func.jac_res(p)
+        return jac[0]
+
+    def setup(self):
+        """
+        Setup problem ready to be run with Kmpfit solver
+        """
+        self.kmpfit_object = kmpfit.Fitter(residuals=self.kmpfit_residuals,
+                                           deriv=self.kmpfit_jacobians,
+                                           data=(self.data_x, self.data_y)
+                                           )
+
+    def fit(self):
+        """
+        Run problem with Kmpfit solver
+        """
+
+        self.kmpfit_object.fit(params0=self.initial_params)
+        self._status = 0 if "success" in self.kmpfit_object.message else 2
```

Comment:
This line should be in cleanup as there's no need to time it.
That way you can also get rid of `self._status`

Reply:
I moved the kmpfit message into the cleanup and the kmpfit controller tests started failing. This was due to `kmpfit_message` object updating to a default value and ignoring or overiding the values defined in the tests.